packages/fx-core/src/component/driver/deploy/azure/baseDeployDriver.ts

```diff
@@ -23,26 +27,25 @@ export abstract class BaseDeployDriver extends BaseDeployStepDriver {
     dryRun: asOptional(asBoolean),
   });
 
-  async run(): Promise<Map<string, string>> {
+  async run(): Promise<ExecutionResult> {
     await this.context.logProvider.debug("start deploy process");
 
-    const deployArgs = BaseDeployDriver.asDeployArgs(this.args, this.helpLink);
-    // if working directory not set, use current working directory
-    deployArgs.workingDirectory = deployArgs.workingDirectory ?? "./";
-    // if working dir is not absolute path, then join the path with project path
-    this.workingDirectory = path.isAbsolute(deployArgs.workingDirectory)
-      ? deployArgs.workingDirectory
-      : path.join(this.workingDirectory, deployArgs.workingDirectory);
-    // if distribution path is not absolute path, then join the path with project path
-    this.distDirectory = path.isAbsolute(deployArgs.distributionPath)
-      ? deployArgs.distributionPath
-      : path.join(this.workingDirectory, deployArgs.distributionPath);
-    this.dryRun = deployArgs.dryRun ?? false;
-    // call real deploy
-    await this.wrapErrorHandler(async () => {
-      await this.deploy(deployArgs);
+    return await this.wrapErrorHandler(async () => {
+      const deployArgs = BaseDeployDriver.asDeployArgs(this.args, this.helpLink);
+      // if working directory not set, use current working directory
+      deployArgs.workingDirectory = deployArgs.workingDirectory ?? "./";
+      // if working dir is not absolute path, then join the path with project path
+      this.workingDirectory = path.isAbsolute(deployArgs.workingDirectory)
+        ? deployArgs.workingDirectory
+        : path.join(this.workingDirectory, deployArgs.workingDirectory);
+      // if distribution path is not absolute path, then join the path with project path
+      this.distDirectory = path.isAbsolute(deployArgs.distributionPath)
```

Comment:
It looks this logic of get final path is the same as "workingDirectory", so is it possible to extract it as a method so that it will looks like
```
this.workingDirectory = resolvePath(deployArgs.workingDirectory);
this.distDirectory = resolvePath(deployArgs.distDirectory);
```

Reply:
Another question is that does "this.workingDirectory" refer to "project path" in the comment?

Reply:
`this.workingDirectory` was set the value to `project path` in `BaseDeployStepDriver`. Because the default working directory is the project path.